### PR TITLE
[POAE7-2453]directly support presto date extract functions 

### PR DESCRIPTION
**Pull request by spevenhe**
### What changes were proposed in this pull request?

directly support presto date extract functions like month(), day(), day_of_year()....

### Why are the changes needed?

support upstream Uts

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

UTs
### Which label does this PR belong to?

CODEREFACTOR

### CODE REVIEW

File: cider/exec/template/DateTimeTranslator.cpp [hunk 1]
```diff
@@ -117,6 +117,53 @@ std::string from_datetrunc_field(const DatetruncField& fieldno) {
 
 }  // namespace
 
+ExtractField ExtractExpr::presto_function_to_extract_field(
```
Comment (jikunshang) [inline cider/exec/template/DateTimeTranslator.cpp:120]: rename to `try_extract_presto_function`?

File: cider/exec/template/DateTimeTranslator.cpp [hunk 2]
```diff
@@ -117,6 +117,53 @@ std::string from_datetrunc_field(const DatetruncField& fieldno) {
...
+    const std::string& function_name) {
+  ExtractField fieldno;
```
Comment (jikunshang) [inline cider/exec/template/DateTimeTranslator.cpp:122]: hmmm what's no in filedno mean?

File: cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp [hunk 3]
```diff
@@ -654,16 +654,26 @@ std::shared_ptr<Analyzer::Expr> Substrait2AnalyzerExprConverter::buildInValuesEx
...
-      toAnalyzerExpr(s_scalar_function.arguments(1).value(), function_map, expr_map_ptr);
-  return ExtractExpr::generate(from_expr, time_unit);
+  if (extract_field == ExtractField::NONE) {
+    CHECK(s_scalar_function.arguments_size() == 2);
+    CHECK(s_scalar_function.arguments(1).has_value());
+    auto from_expr = toAnalyzerExpr(
+        s_scalar_function.arguments(1).value(), function_map, expr_map_ptr);
+    CHECK(s_scalar_function.arguments(0).has_enum_());
+    std::string time_unit = s_scalar_function.arguments(0).enum_().specified();
+    CHECK(time_unit != *&::google::protobuf::internal::GetEmptyStringAlreadyInited());
```
Comment (jikunshang) [inline cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp:667]: emmm can igonre `*&`? seems hard to understand.

File: cider/exec/template/DateTimeTranslator.cpp [hunk 4]
```diff
@@ -117,6 +117,53 @@ std::string from_datetrunc_field(const DatetruncField& fieldno) {
...
+    fieldno = kQUARTERDAY;
+  } else if (boost::iequals(function_name, "hour")) {
+    fieldno = kHOUR;
+  } else if (boost::iequals(function_name, "minute")) {
+    fieldno = kMINUTE;
+  } else if (boost::iequals(function_name, "second")) {
+    fieldno = kSECOND;
+  } else if (boost::iequals(function_name, "millisecond")) {
+    fieldno = kMILLISECOND;
+  } else if (boost::iequals(function_name, "microsecond")) {
```
Comment (kd97) [inline cider/exec/template/DateTimeTranslator.cpp:141]: use map better?

File: cider/exec/template/DateTimeTranslator.cpp (hunk 2 above)
Comment (spevenhe) [reply cider/exec/template/DateTimeTranslator.cpp:122]: field_Number

File: cider/exec/template/DateTimeTranslator.cpp (hunk 4 above)
Comment (spevenhe) [reply cider/exec/template/DateTimeTranslator.cpp:141]: I previously considered changing input to lower case and use a map to compare. But I see lots of large casewhen, if in our code, but I haven't seen somewhere using a map.

Comment (kd97) [reply cider/exec/template/DateTimeTranslator.cpp:141]: just a suggestion. I read some codes. Ref
https://github.com/intel/BDTK/blob/af0dd42b977332488f33cf1b0984a606707a78e3/cider/exec/template/DateTimeUtils.h#L38-L49

File: cider/exec/template/DateTimeTranslator.cpp (hunk 1 above)
Comment (spevenhe) [reply cider/exec/template/DateTimeTranslator.cpp]: updated

File: cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp (hunk 3 above)
Comment (spevenhe) [reply cider/exec/plan/parser/SubstraitToAnalyzerExpr.cpp]: updated

File: cider/function/datetime/ExtractFromTime.h [hunk 5]
```diff
@@ -95,7 +95,8 @@ enum ExtractField {
   kWEEK,
   kWEEK_SUNDAY,
   kWEEK_SATURDAY,
-  kDATEEPOCH
+  kDATEEPOCH,
+  NONE
```
[the remark this answers is not in the document]
Comment (spevenhe) [reply cider/function/datetime/ExtractFromTime.h]: updated

File: cider/exec/template/DateTimeTranslator.cpp (hunk 4 above)
Comment (spevenhe) [reply cider/exec/template/DateTimeTranslator.cpp]: updated

File: cider/exec/template/DateTimeTranslator.cpp [hunk 6]
```diff
@@ -115,8 +115,39 @@ std::string from_datetrunc_field(const DatetruncField& fieldno) {
...
+    {"nanosecond", kNANOSECOND},
+    {"day_of_week", kDOW},
+    {"isodow", kISODOW},
+    {"day_of_year", kDOY},
+    {"epoch", kEPOCH},
+    {"week", kWEEK},
+    {"week_sunday", kWEEK_SUNDAY},
+    {"week_saturday", kWEEK_SATURDAY},
+    {"dateepoch", kDATEEPOCH}};
+
```
Comment (yma11) [inline cider/exec/template/DateTimeTranslator.cpp:138]: remove empty line?

Comment (spevenhe) [reply cider/exec/template/DateTimeTranslator.cpp:138]: next line is 

```}  // namespace```

should there be no extra line?

**Review by jikunshang — APPROVED** [review]

**Pull request merged**